Cleanup and Testing

ZachPeterson opened the pull request:
Cleaned up a few things with the database and models. Added some default testing values to be inserted into the database if in testing mode.

ZachPeterson committed:
Cleanup and Testing

- Added .vscode to the .gitignore and removed it from the project.
- Added SECRET_KEY to application config and set testing for development
config to true.
- Removed unneeded imports from Authentication.py and updated
capitalization of API to match other classes and packages.
- Moved email based id listing left over from old system to integer
based id listing.
- Made methods of API static as they do not need access to an API
instance and fixed queries to use proper ids.
- Updated DBModels to use regular integer primary keys, which triggers
SQLite to use a special unique table row system as described at:
http://docs.sqlalchemy.org/en/latest/dialects/sqlite.html. Also updated
naming scheme to match our site terminology (moved lecture to class as
lectures will be used to store groups of questions) and changed table
names to be plural (a table holds a collection of 'Users' for example,
but the model operates on single 'User' objects).
- Added additional styling to our hosted content, rather than CDN, so
that we have full access to the unminifed versions for debugging as well
as being able to run the full application entirely offline.
- Updated naming of student lectures in the Angular controller to
reflect the change from lecture->class. Also fixed response.data for
set_student_classes checking the wrong type and not successfully adding
a class as a result.
- Updated layout with script and style resources for self hosting.
- Updated class list view to use class list controller instead of
lectures controller.
- Added database test content to app initialization. If TESTING is set
to true for the current application content, then the database contents
will be DROPPED and recreated and filled with fresh test data. This
should allow us to have a fully functioning app with reasonable test
data for the start of every demo.

ZachPeterson committed:
Moved API from api.py to API.py

goutamvenkat commented:
can we change the professor url request to not return a string version of bool? string comparisons are too slow

goutamvenkat commented on ClassMoodApp/Controllers/ClassList.py:
```diff
@@ -1,39 +1,34 @@
     if successful:
         return 'True'
     else:
         return 'False'
```
This is unnecessary, instead use `return json.dumps(successful)`

goutamvenkat commented on ClassMoodApp/static/js/controllers.js:
```diff
@@ -73,9 +74,9 @@ app.controller('dialogController', function($scope, $log, $http) {
+            $http.post('/set_student_classes/' + className + '/' + studentID)
             .then(function successfulCallback(response) {
-                if (response.data === true) {
+                if (response.data == 'true') {
```
No this is incorrect, don't use string comparison. Update the API and this function

ZachPeterson replied:
Unfortunately this is correct. Angular documentation shows that the response data is held as a string (https://docs.angularjs.org/api/ng/service/$http) and logging the typeof response.data confirms this. We can either do a string comparison here or use the transform method to convert the string response from the server to a boolean.

goutamvenkat commented on ClassMoodApp/static/js/controllers.js:
```diff
@@ -49,10 +50,10 @@ app.controller('dialogProfessorController', function($scope, $log, $http) {
-app.controller("studentLectures", function($scope, $http) {
+app.controller("studentClassListController", function($scope, $http) {
     // Get studentID from session token
     var studentID = 1;
```
studentID is hardcoded. Needs to be retrieved from cookie

goutamvenkat commented on ClassMoodApp/Models/DBModels.py:
```diff
@@ -9,80 +9,77 @@
-    __tablename__ = 'Roster'
-    student_id = db.Column(db.BigInteger(), db.ForeignKey('User.id'), primary_key=True)
-    lecture_id = db.Column(db.BigInteger(), db.ForeignKey('Lecture.id'), primary_key=True)
+class ClassMember(db.Model):
```
Why was this table name changed??? Roster made more sense

ZachPeterson replied:
Table name changed to reflect the name agreed upon by the team during the database architecture meeting.

goutamvenkat reviewed (changes requested):
The changes I mentioned should be addressed before merging. Personally, I don't see why the table names were changed, but whatever. Regardless, there some places were there's no underscore casing, those should be fixed for consistent stylistic purposes. The main thing to change though is the api function that returns successful.

Besides that it seems fine.

goutamvenkat commented:
In login page, material design lite doesn't work and also a bunch of js files don't load. 
![Screenshot of console](https://cloud.githubusercontent.com/assets/5576941/18832353/f6250a4c-83b8-11e6-858e-cfff7898073c.png)

goutamvenkat commented:
Getting this after logging in: 
![After login image](https://cloud.githubusercontent.com/assets/5576941/18832613/84680f74-83ba-11e6-9f3d-37a061b8c7fa.png)

goutamvenkat commented:
Once I'm logged in and try hitting localhost again, it sends me to this page
![image](https://cloud.githubusercontent.com/assets/5576941/18832743/333106e6-83bb-11e6-8cc1-c4a3bca1ca85.png)

which makes absolutely no sense. The logout should be placed in each of the views (student and professor). If the user is authenticated, why are we redirecting to authtest.html?

ZachPeterson committed:
Adding Library Files

Lib was in the gitignore so the new angular offline files were not added
properly

krabagobanjo reviewed (changes requested):
The authentication redirect is a known issue - I didn't want to screw with the UI for the professor and student pages but we needed some way to log out.  The framework is there for implementation and should be done for next sprint.

I attached one comment, but everything seems fine otherwise.

ZachPeterson committed:
Updated Return Type

Updated the return type of create_prof_class to us json.dumps
